node/router/src/heartbeat.rs

```diff
@@ -104,8 +107,49 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
         }
     }
 
-    /// This function removes the oldest connected peer, to keep the connections fresh.
-    /// This function only triggers if the router is above the minimum number of connected peers.
+    /// Returns a sorted list of removable connected peers
+    /// where the first entry has the lowest priority and
+    /// the last one the highest.
+    ///
+    /// Rules:
+    ///     - Trusted peers and bootstrap nodes are not removable
+    ///     - Peers that we are currently syncing with are not removable
+    ///     - Validators are considered higher priority than clients
```

Comment:
I think this comment (or the beginning of this doc comment) might be out of date w.r.t. the code as the current implementation sorts validators at the front of the returned vector (and from most recently seen to least recently seen)?

Reply:
It now sorts by `(peer.is_validator(), peer.last_seen())`, so older peers and clients should have lower priority as described in the comment. 
Let me know if you still think there is an inconsistency.